smithy-linters/src/main/java/software/amazon/smithy/linters/CamelCaseValidator.java

```diff
@@ -135,11 +151,27 @@ public List<ValidationEvent> validate(Model model) {
                         shape.getType(), shape.getId().getName())))
                 .forEach(events::add);
 
-        Pattern isValidMemberName = config.getMemberNames().getRegex();
+        Pattern memberNamePattern;
+        if (MemberNameHandling.AUTO.equals(config.getMemberNames())) {
+            long upperMemberNameCount = model.shapes(MemberShape.class)
+                    .filter(shape -> !shape.getContainer().getNamespace().equals(Prelude.NAMESPACE))
+                    .filter(shape -> MemberNameHandling.UPPER.getRegex().matcher(shape.getMemberName()).find())
+                    .count();
+            long lowerMemberNameCount = model.shapes(MemberShape.class)
+                    .filter(shape -> !shape.getContainer().getNamespace().equals(Prelude.NAMESPACE))
+                    .filter(shape -> MemberNameHandling.LOWER.getRegex().matcher(shape.getMemberName()).find())
+                    .count();
```

Comment:
It would be better to use a single for-loop here. You can grab a cached set of members with `model.toSet(MemberShape.class)`. You're also going to want to filter out the following namespaces:

* `aws.apigateway`
* `aws.cloudformation`
* `aws.iam`
* `aws.api`
* `aws.auth`
* `aws.customizations`
* `aws.protocols`
* `smithy.test`